### ISSUE TITLE
Scroll to item as well

### ISSUE THREAD

**Issue by amahdy** [vaadin/vaadin-grid#1154]
### Description
I want to be able to `scrollToItem` in case I don't have the index. Or maybe `getItemIndex`.

### Expected outcome
Being able to scroll without keeping track of indexes.

### Actual outcome
Currently I must keep track of indexes of all items.

**Comment by tomivirkki** [comment]
This isn't logically possible since the grid can't know the presentational index of an arbitrary item in case a `dataProvider` is used. This would be doable for items that happen to be cached by the grid at a certain time or for the `items` array as when it's used, grid knows all the items and their indexes beforehand, but we would want to avoid adding API that only works under some specific condition, especially since scrolling to an index works regardless of the grid's state.

**Comment by amahdy** [comment]
Not sure if this can make sense but maybe we can have some `itemByIndexHandler` in case of `dataProvider`. Pretty much like how `size` is required when using `dataProvider`, `getItemIndex` would return the item from the cached array or the overridden implementation in case of `dataProvider`.

**Comment by wuwu2000** [comment]
One year later this still is bugging me. This issue is leading to very poor UX.
I think you can  make 95% happy by implementing a method scrollToSelectedItem(). That should be an implementation without needing some fancy handler or so. In the end it is just an javascript execution I hope. If there is multiselection enabled, just scroll to the first selected item.
Thanks for reading and happy holidays

**Comment by tomivirkki** [comment]
Hi, I still can't see how this could work in practice. Let's say your database has 1M items and you haven't yet scrolled the grid so it has only loaded the first pages of data. 

Then you set an item selected with `grid.selectItem(item)` where the `item` is some previously stored selection (at database index 800 000, so it's not one of the items already loaded by grid). Because grid didn't requested the item, it doesn't know the index either, right.

Invoking `grid.scrollToSelectedItem()` should in theory then scroll to the item but there's no way that grid would know that the item is actually at index 800 000 so it doesn't know where to scroll...

Happy holidays

**Comment by johnthad** [comment]
I agree that scrolling beyond the current loaded values is not practical. But what about an API call to scroll to an item within the current `items` property (type `Array`) by passing an index number? If there are 50 items loaded, scrolling beyond 0-49 would throw an error (or maybe an index of `-1` could be shorthand for the last item).

In `vaadin-grid` 4.0.0, when I needed to scroll to the first item (a record in a search hitlist), I used `this.$.grid._scrollToIndex(0)`, knowing that `_scrollToIndex(_number_)` was a "private" method. I'm starting today or tomorrow to upgrade that component. I'd like the same capability, and for that capability to stick around as `vaadin-grid` improves.

**Comment by TatuLund** [comment]
This has been on a wis list since Vaadin 8 was released. Here is an old ticket, where I have expressed my ideas how to do this https://github.com/vaadin/framework/issues/9266#issuecomment-455287292

My five cents on this

- Add getIndexOf(T item) to DataProvider interface

- Add setter for index provider function to DataProvider interface

- Add trivial implementation of the above in ListDataProvider

- Add implementation of scrollTo(T item) that relies on above, ensure that IllegalStateException is thrown if index provider function is not available

- It is responsibility of application developer to provide efficient and correct index provider function

**Comment by tomivirkki** [comment]
> Add implementation of scrollTo(T item) that relies on above

This would just be a convenience API for not that inconvenient:
```
grid.scrollToIndex(dataProvider.getIndexOf(item));
```

**Comment by mstahv** [comment]
The academically correct place in master branch might be the GridListDataView class. One now gets a reference to that when using setItems method with an array or collection, in which cases it would be rather cheap to detect the correct index.

**Comment by DykiSA** [comment]
Is there any updates on this? I am looking forward for the simple workaround
